### PR TITLE
Create methods for getting and setting a config

### LINKED ISSUE #418: Create methods for getting and setting a config
### Problem Description

The HyperTransformer's behavior is determined by the `sdtypes` and `transformers` -- called the config. Provide methods to get and set these.

### Expected behavior

Implement a `get_config()` method that returns the `sdtypes` and `transformers` for each of the column names.

```python
{
  'sdtypes': {
    'colA': 'datetime',
    'colB': 'categorical'
  },
  'transformers': {
    'colA': <UnixTimestampEncoder(missing_value_replacement="mean")>,
    'colB': <FrequencyEncoder()>
  }
}
```

Implement a `set_config()` method that accepts a param (`config`) where the user can specify this themselves. Note that this method resets the entire config. If a user wishes to modify parts of it, they must use the `update_sdtypes` or `update_transformers` methods instead.

```python
ht.set_config(config={
  'sdtypes': {
    'colA': 'datetime',
    'colB': 'categorical'
  },
  'transformers': {
    'colA': UnixTimestampEncoder(missing_value_replacement="mean"),
    'colB': FrequencyEncoder()
  }
})
```

While we're at it, we can also remove any existing getters for either the sdtypes or transformers alone.

### Edge Cases

A user may be attempting to set a transformer that is incompatible with a particular sdtype. Throw a warning for each incompatible column.

```python
ht.set_config(config={
  'sdtypes': {
    'last_login': 'datetime'
  },
  'transformers': {
    'last_login': FloatFormatter(missing_value_replacement="mean")
  }
})
Warning: You are assigning a numerical transformer to a datetime column ('last_login').
If the transformer doesn't match the sdtype, it may lead to errors.
```

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #418

### CODE REVIEW

File: rdt/hyper_transformer.py [hunk 1]
```diff
@@ -162,6 +162,45 @@ def _unfit(self):
...
         self._fitted = False
 
+    @staticmethod
+    def _validate_config(config):
+        sdtypes = config.get('sdtypes', {})
+        transformers = config.get('transformers', {})
+        for column, transformer in transformers.items():
+            input_type = transformer.get_input_type()
+            sdtype = sdtypes.get(column)
+            if transformer.get_input_type() != sdtypes.get(column):
```
Comment (fealho) [inline rdt/hyper_transformer.py:172]: Do `if input_type != sdtype:` instead.

Comment (fealho) [inline rdt/hyper_transformer.py:172]: Do `if input_type != sdtype:` instead.

**Review by fealho — COMMENTED** [review]
Looks good!  Only thing remaining is to do the "remove any existing getters for either the sdtypes or transformers alone." part of the issue.

**Review by fealho — APPROVED** [review]

File: rdt/hyper_transformer.py [hunk 2]
```diff
@@ -162,15 +162,44 @@ def _unfit(self):
...
-        """Get the ``field_data_types`` dict.
+    @staticmethod
+    def _validate_config(config):
+        sdtypes = config.get('sdtypes', {})
+        transformers = config.get('transformers', {})
+        for column, transformer in transformers.items():
+            input_type = transformer.get_input_type()
+            sdtype = sdtypes.get(column)
+            if input_type != sdtype:
+                warnings.warn(f'You are assigning a {input_type} transformer to a {sdtype} column'
```
Comment (pvk-developer) [inline rdt/hyper_transformer.py:173]: Should we then change all the `prints` that `warn` to warnings ? I'm getting a little bit confused with `print` and `warning`.

**Review by pvk-developer — APPROVED** [review]

**Pull request merged**